In [2]:
import yfinance as yf

In [3]:
# Define the stock ticker and time period
ticker = 'AAPL'   # Apple Inc.
start_date = '2018-01-01'   # Start date for the data
end_date = '2023-01-01'     # End date for the data

In [4]:
# Download stock data
stock_data = yf.download(tickers=ticker, start=start_date, end=end_date)

[*********************100%***********************]  1 of 1 completed


In [5]:
len(stock_data)

1259

In [6]:
stock_data.head(20)

Open       High        Low      Close  Adj Close     Volume
Date                                                                        
2018-01-02  42.540001  43.075001  42.314999  43.064999  40.568920  102223600
2018-01-03  43.132500  43.637501  42.990002  43.057499  40.561859  118071600
2018-01-04  43.134998  43.367500  43.020000  43.257500  40.750263   89738400
2018-01-05  43.360001  43.842499  43.262501  43.750000  41.214230   94640000
2018-01-08  43.587502  43.902500  43.482498  43.587502  41.061150   82271200
2018-01-09  43.637501  43.764999  43.352501  43.582500  41.056435   86336000
2018-01-10  43.290001  43.575001  43.250000  43.572498  41.047012   95839600
2018-01-11  43.647499  43.872501  43.622501  43.820000  41.280167   74670800
2018-01-12  44.044998  44.340000  43.912498  44.272499  41.706440  101672400
2018-01-16  44.474998  44.847500  44.035000  44.047501  41.494480  118263600
2018-01-17  44.037498  44.812500  43.767502  44.775002  42.179829  137547200
2018-01-18  44.842499  45.025002  44.562500  44.814999  42.217495  124773600
2018-01-19  44.652500  44.895000  44.352501  44.615002  42.029099  129700400
2018-01-22  44.325001  44.445000  44.150002  44.250000  41.685246  108434400
2018-01-23  44.325001  44.860001  44.205002  44.259998  41.694660  130756400
2018-01-24  44.312500  44.325001  43.299999  43.555000  41.030537  204420400
2018-01-25  43.627499  43.737499  42.632500  42.777500  40.298088  166116000
2018-01-26  43.000000  43.000000  42.514999  42.877499  40.392292  156572000
2018-01-29  42.540001  42.540001  41.767502  41.990002  39.556240  202561600
2018-01-30  41.382500  41.842499  41.174999  41.742500  39.323082  184192800

In [8]:
# Save the stock data to a csv file
stock_data.to_csv(f'../data/raw/{ticker}_stock_data.csv')

## Data Collection Summary

In this phase, a historical stock data was collected from Yahoo Finance using the `yfinance` library. Specifically, 1,259 daily data points were obtained by downloading Apple Inc. (AAPL) stock prices between January 1, 2018, and January 1, 2023.

The data includes the following features:
- **Date:** The trading date
- **Open:** The price at market open
- **High:** The highest price during the trading day
- **Low:** The lowest price during the trading day
- **Close:** The price at market close
- **Volume:** The number of shares traded
- **Adj Close:** Adjusted closing price for splits/dividends

The dataset was saved as a `csv` file.